ton_client/src/abi/encode_message.rs

```diff
@@ -333,18 +339,25 @@ pub async fn encode_message(
 ) -> ClientResult<ResultOfEncodeMessage> {
     let abi = params.abi.json_string()?;
 
-    let public = params.signer.resolve_public_key(context.clone()).await?;
-    let (message, data_to_sign, address) = if let Some(deploy_set) = params.deploy_set {
+    let deploy_set = params.deploy_set.as_ref().map(
+        |deploy_set|
+            create_tvc_image(
+                &abi,
+                deploy_set.initial_data.as_ref(),
+                &deploy_set.tvc,
+            ).map(|image| (deploy_set, image))
```

Comment:
Public key resolved by this function should be used only in tvc image. `encode_deploy` and `encode_run` functions should use public key from signer as they did. Call `resolve_pubkey` inside `create_tvc_image`. Add test for such case